homeassistant/components/econet/binary_sensor.py

```diff
@@ -40,7 +44,10 @@ async def async_setup_entry(hass, entry, async_add_entities):
     all_equipment = equipment[EquipmentType.WATER_HEATER].copy()
     all_equipment.extend(equipment[EquipmentType.THERMOSTAT].copy())
     for _equip in all_equipment:
-        for name, attribute in SENSOR_NAMES_TO_ATTRIBUTES.items():
+        attributes = []
+        for sensor in SENSORS.values():
+            attributes.append(sensor[ATTR])
+        for name, attribute in attributes:
```

Comment:
I'm not following here. The attribute is a single string.

Reply:
I think this is right? Needs to move out of the for loop thought, no need to calculate it every time.

Basically collect all of the possible attributes and then check each EcoNet device for that attribute.

Reply:
No, `sensor[ATTR]` will return a value which is just the string representing the attribute.

Reply:
Correct, and then on line 51 we check the equipment for that attribute using getattr. It does seems like someone broke with this change. I will need to try to test things locally.

Reply:
The `attributes` variable is then a list with just the attribute strings. Ie we can't do a tuple unpack like we do now.